ClearentIdtechIOSFramework/ClearentUI/ClearentProcessingModalViewController.swift

```diff
@@ -45,37 +62,80 @@ extension ClearentProcessingModalViewController: ClearentProcessingModalView {
+            var readersTableViewDataSource: [ReaderItem] = readersInfo.map { item -> ReaderItem in
+                    return ReaderItem(readerInfo: item)
+            }
+            
+            if !pairedReaderInfo.isConnected {
+                guard let selectedReaderFromReadersList = presenter?.getSelectedReaderFromReadersList() else {
+                    return ClearentReadersTableView(dataSource: readersTableViewDataSource, delegate: self)
+                }
+                guard let indexOfSelectedReader = readersTableViewDataSource.firstIndex(where: {$0.readerInfo.readerName == selectedReaderFromReadersList.readerInfo.readerName}) else { return nil }
+                readersTableViewDataSource[indexOfSelectedReader].isConnecting = true
+                
+                return ClearentReadersTableView(dataSource: readersTableViewDataSource, delegate: self)
+            } else {
+                guard let indexOfConnectedReader = readersTableViewDataSource.firstIndex(where: {$0.readerInfo.readerName == pairedReaderInfo.readerName}) else { return nil }
+                readersTableViewDataSource.insert(readersTableViewDataSource.remove(at: indexOfConnectedReader), at: 0)
+                presenter?.setSelectedReaderFromReadersList(nil)
+                
+                return ClearentReadersTableView(dataSource: readersTableViewDataSource, delegate: self)
+            }
         }
     }
 
     private func readerInfoView(readerInfo: ReaderInfo, flowFeedbackType: FlowFeedbackType) -> ClearentReaderStatusHeaderView {
         let name = readerInfo.readerName
-        let signalStatus = readerInfo.signalStatus(flowFeedbackType: flowFeedbackType)
+        let signalStatus: (iconName: String?, title: String)
+        
+        if let _ = presenter?.getSelectedReaderFromReadersList() {
+            signalStatus = readerInfo.signalStatus(flowFeedbackType: flowFeedbackType, isConnecting: true)
+        } else {
+            signalStatus = readerInfo.signalStatus(flowFeedbackType: flowFeedbackType)
+        }
         let batteryStatus = readerInfo.batteryStatus(flowFeedbackType: flowFeedbackType)
         let statusHeader = ClearentReaderStatusHeaderView()
-        statusHeader.setup(readerName: name, signalStatusIconName: signalStatus.iconName, signalStatusTitle: signalStatus.title, batteryStatusIconName: batteryStatus.iconName, batteryStatusTitle: batteryStatus.title)
+        
+        showOnTop ? statusHeader.setup(readerName: name, dropDownIconName: ClearentConstants.IconName.expanded, signalStatusIconName: signalStatus.iconName, signalStatusTitle: signalStatus.title, batteryStatusIconName: batteryStatus.iconName, batteryStatusTitle: batteryStatus.title) : statusHeader.setup(readerName: name, signalStatusIconName: signalStatus.iconName, signalStatusTitle: signalStatus.title, batteryStatusIconName: batteryStatus.iconName, batteryStatusTitle: batteryStatus.title)
```

Comment:
This line is hard to read. Could be simplified like this.
```suggestion
let dropDownIconName = showOnTop ? ClearentConstants.IconName.expanded : nil
statusHeader.setup(readerName: name, dropDownIconName: dropDownIconName, signalStatusIconName: signalStatus.iconName, signalStatusTitle: signalStatus.title, batteryStatusIconName: batteryStatus.iconName, batteryStatusTitle: batteryStatus.title)
```
You can also set fewer parameters to the `setup` method: only `readerName`, `dropDownIconName`, `signalStatus` & `batteryStatus`